core/src/main/scala/tapir/Endpoint.scala

```diff
@@ -116,9 +116,67 @@ case class Endpoint[I, E, O, +S](input: EndpointInput[I], errorOutput: EndpointO
     */
   def showRaw: String = toString
 
+  /**
+    * Renders endpoint path, by default all parametrised path and query components are replaced by {param_name} or {paramN}, e.g. for
+    * {{{
+    * endpoint.in("p1" / path[String] / query[String]("par2"))
+    * }}}
+    * returns `/p1/{param1}?par2={par2}`
+    */
+  def renderPath(
+      renderPathComponent: (Int, PathCapture[_]) => String = Endpoint.defaultPathCaptureRender,
+      renderQueryComponent: Option[(Int, Query[_]) => String] = Some(Endpoint.defaultQueryRender)
+  ): String = {
+    import tapir.internal._
+
+    def namedPathComponents(
+        inputs: Vector[EndpointInput.Basic[_]],
+        renderPath: (Int, PathCapture[_]) => String
+    ): (Vector[String], Int) =
+      inputs.foldLeft((Vector.empty[String], 1)) {
+        case ((acc, index), component) =>
+          component match {
+            case p: EndpointInput.PathCapture[_] => (acc :+ renderPath(index, p), index + 1)
+            case EndpointInput.FixedPath(s)      => (acc :+ s, index)
+            case _                               => (acc, index)
+          }
+      }
+
+    def namedQueryComponents(
+        inputs: Vector[EndpointInput.Basic[_]],
+        renderQuery: (Int, Query[_]) => String,
+        pathParamCount: Int
+    ): Vector[String] =
+      inputs
+        .foldLeft((Vector.empty[String], pathParamCount)) {
+          case ((acc, index), component) =>
+            component match {
+              case q: EndpointInput.Query[_] => (acc :+ renderQuery(index, q), index + 1)
+              case _                         => (acc, index)
+            }
+        }
+        ._1
+
+    val inputs = input.asVectorOfBasicInputs(includeAuth = false)
```

Comment:
I suppose we should include `Auth` inputs as well, as they might wrap path/query params?

Reply:
Only `Auth.ApiKey` is capable of doing that, right?

Reply:
In theory you can create an instance of any of the `Auth` implementations containing an arbitrary `EndpointInput.Single`

Reply:
@adamw I was still intending to look at it :P New PR then? ;)

Reply:
ah :D forgot about that. Yes, a separate PR please, or I'll try to fix it on Monday :)

Reply:
Definitely won't find any time before middle of the week though.

Reply:
We'll see who's first ;-)